### ISSUE TITLE
conditional group is not reactive on first change

### ISSUE THREAD

**Issue by alwex** [wearebraid/vue-formulate#117]
**Describe the bug**
I am working on a conditional rendered shipping form using groups. The `address` group is conditioned based on the country code and render a different set of field on change.

```
<template>
 <FormulateForm v-model="fields">
   <FormulateInput type="group" name="address">
      <AddressFormUS v-if="country === 'US'" />
      <AddressFormZA v-else-if="country === 'ZA'" />
      <AddressFormDefault v-else />
    </FormulateInput>
  </FormulateForm>
</template>
```

the data structure is
```
data() {
  return {
    fields: {
      address: {
        country: this.country // get this from the props
      }
    }
  }
}
```

when country change, the correct form part is displayed, each form part contains approximatively the same data, just add some extra fields for specific countries

example of addressForm:
```
<div>
    <FormulateInput
          type="text"
          name="line2"
          label="Complex / Building (Optional)"
          validation="required"
          error-behavior="submit"
     />
    <!-- More fields -->
</div>
```

The first change of country does initialise the address form data correctly, all the subsequent changes works perfectly fine.

Here are some screen captures:

1. first render with `country == 'US'`
![image](https://user-images.githubusercontent.com/63174/84467023-f57be780-acce-11ea-9303-aaa0c0ccccce.png)

1. switching to another country `country == 'FR'`
![image](https://user-images.githubusercontent.com/63174/84467061-0af11180-accf-11ea-8634-27a08c2cb79d.png)
all fields are empty

1. switching to another country `country == 'ZA'`
![image](https://user-images.githubusercontent.com/63174/84467087-1f350e80-accf-11ea-84d1-51fc1a689bab.png)
everything is back to normal and fields are correctly populated. From here, switching between countries works correctly, only the first country change cause an issue.

**Device information:**
 - Device: Macbook  Pro
 - OS: MacOS
 - Browser Chrome
 - Version 83.0.4103.61

**Comment by justin-schroeder** [comment]
Hi @alwex! Thanks for the screenshots and detailed write up. Looks like a really nice address form. I'm following what you've got going on, but I'm not 100% sure what problem is? Maybe you could just rephrase that part of the issue for me?

Other than that — 2 quick things that might be causing trouble:

1. The schema for groups is always an array. So where your model is:

```js
data() {
  return {
    fields: {
      address: {
        country: this.country // get this from the props
      }
    }
  }
}
```

It should be:

```js
data() {
  return {
    fields: {
      address: [{
        country: this.country // get this from the props
      }]
    }
  }
}
```

It's a small subtle difference, but important for proper hydration. This allows formulate users to switch between single and repeatable without changing their schema at all.

2. When a field is removed from a formulate form or a group it removes its value automatically, so if you were trying to persist data between the two it would be an issue. Not sure if thats a concern for your use case.

If it's helpful to show the issue here's a codepen template with Vue Formulate already installed:

https://codepen.io/boyd/pen/YzyJPor

**Comment by alwex** [comment]
Hi @justin-schroeder, thank you for your answer. The array for group is just a typo I have introduced on the github issue. I don't have this issue in my code.
I have reproduced the issue in a codepen, just try to switch between the countries, you will notice that the behaviour is somewhat strange, country code ends up being in line2 when switching from US to FR and the select input does not update to FR.

This issues seems to happened when the subform does not share the same inputs. Could be caused by the group input.

here is the codepen

https://codepen.io/alwex-the-bashful/pen/mdVPNJp
